dex/errors_test.go

```diff
@@ -0,0 +1,45 @@
+package dex
+
+import (
+	"errors"
+	"testing"
+)
+
+type testErr string
+
+func (te testErr) Error() string {
+	return string(te)
+}
+
+const (
+	err0 = testErr("other error")
+	err1 = ErrorKind("error 1")
+	err2 = ErrorKind("error 2")
+)
```

Comment:
Illustrating the purpose of `ErrorKind` (to avoid the boiler plate of creating a type that defines `Error() string`), while allowing a `const`, which `errors.New` prevents.